hibernate-core/src/main/java/org/hibernate/boot/model/internal/GeneratorBinder.java

```diff
@@ -560,23 +560,22 @@ private static void checkVersionGenerationAlways(MemberDetails property, Generat
 
 	/**
 	 * If the given {@link Generator} also implements {@link Configurable},
-	 * call its {@link Configurable#configure(Type, Properties, ServiceRegistry)
+	 * call its {@link Configurable#configure(Type, Properties, GeneratorCreationContext)
 	 * configure()} method.
 	 */
 	private static void callConfigure(
 			GeneratorCreationContext creationContext,
 			Generator generator,
 			Map<String, Object> configuration,
 			SimpleValue identifierValue) {
-		if ( generator instanceof Configurable ) {
-			final Configurable configurable = (Configurable) generator;
+		if ( generator instanceof final Configurable configurable ) {
```

Comment:
Are we doing flow-typing now?

If so, damn, there's a *whole lot* of code that is going to look a *lot* prettier!

Reply:
With Hibernate 7 we moved to JDK 17 baseline, so we can indeed use pattern matching for `instanceof`.